### PR TITLE
Validate the item properties for made up alpha grid

### PR DESCRIPTION
**Pull request by vigneshvg**
Also address the comments from the review here:
https://aomedia-review.googlesource.com/c/libavif/+/175421

### CODE REVIEW

**Comment by vigneshvg** [comment]
@wantehchang this PR addresses the review comments in https://aomedia-review.googlesource.com/c/libavif/+/175421.

I will add the unit test in a separate PR since that might involve adding a new test file and updating the cmake files, etc.

**Comment by vigneshvg** [comment]
Unit test is being added here: https://github.com/AOMediaCodec/libavif/pull/1402

File: src/read.c [hunk 1]
```diff
@@ -3588,27 +3588,30 @@ static avifDecoderItem * avifDecoderDataFindColorItem(avifDecoderData * data)
...
     const avifProperty * auxCProp = avifPropertyArrayFind(&item->properties, "auxC");
     return auxCProp && isAlphaURN(auxCProp->u.auxC.auxType);
 }
 
 // Finds the alpha item whose parent item is colorItem and sets it in the alphaItem output parameter. Returns AVIF_RESULT_OK on
 // success. Note that *alphaItem can be NULL even if the return value is AVIF_RESULT_OK. If the colorItem is a grid and the alpha
-// item is represented as a set of auxl items to each color tile, then a fake item will be created and isAlphaItemInInput will be
-// set to AVIF_FALSE. Otherwise, isAlphaItemInInput will be set to AVIF_TRUE when *alphaItem is not NULL.
+// item is represented as a set of auxl items to each color tile, then a fake item will be created and *isAlphaItemInInput will be
+// set to AVIF_FALSE. In this case, the alpha item merely exists as a pointer to hold the locations of the alpha grid items. The
```
Comment (wantehchang) [inline src/read.c:3604]: 1. Nit: The word "pointer" is a little unclear here. Maybe we can remove "as a pointer"?

2. alpha grid items => alpha tile items ?

File: src/read.c [hunk 2]
```diff
@@ -3588,27 +3588,30 @@ static avifDecoderItem * avifDecoderDataFindColorItem(avifDecoderData * data)
...
+// data of this item need not be read and the properties need not be validated. Otherwise, *isAlphaItemInInput will be set to
```
Comment (wantehchang) [inline src/read.c:3605]: Delete "and the properties need not be validated", or updated it to say the pixi property cannot be validated.

**Review by wantehchang — APPROVED** [review]
LGTM. Thanks!

I suggest some tweaks to the comments.

File: src/read.c (hunk 1 above)
Comment (vigneshvg) [reply src/read.c:3604]: Done.

File: src/read.c (hunk 2 above)
Comment (vigneshvg) [reply src/read.c]: Done.

File: src/read.c [hunk 3]
```diff
@@ -3977,11 +3982,15 @@ avifResult avifDecoderReset(avifDecoder * decoder)
...
             avifDecoderItemValidateProperties(colorItem, avifGetConfigurationPropertyName(colorCodecType), &decoder->diag, decoder->strictFlags));
-        if (alphaItem && isAlphaItemInInput) {
-            AVIF_CHECKRES(avifDecoderItemValidateProperties(alphaItem,
-                                                            avifGetConfigurationPropertyName(alphaCodecType),
-                                                            &decoder->diag,
-                                                            decoder->strictFlags));
+        if (alphaItem) {
+            avifStrictFlags strictFlags = decoder->strictFlags;
+            if (!isAlphaItemInInput) {
+                // In this case, the made up grid item will not have a an associated pixi property. So validate everything else
```
Comment (wantehchang) [inline src/read.c:3988]: Nit: Remove "a" from "a an associated".

**Review by wantehchang — COMMENTED** [review]
LGTM.

File: src/read.c (hunk 3 above)
Comment (wantehchang) [reply src/read.c:3988]: Fixed in https://github.com/AOMediaCodec/libavif/pull/1406.

**Pull request merged**